frontend/javascript/README.md

```diff
@@ -0,0 +1,713 @@
+    static defaultName = "Animal";
+    // private properties start with a hash (#) and cannot be accessed outside
+    // of the class
+    #message;
+
+    // methods are also defined with just a name
+    sayMessage() {
+        return `${name !== undefined ? this.name : Animal.defaultName} says: ${this.#message}`;
+    }
+
+    // private methods also start with a hash (#). they can also be static.
+    #getNameOrDefault() {
+        return name !== undefined ? this.name : Animal.defaultName;
+    }
+
+    // constructors give you a way of creating an object from the class
+    // blueprint. the constructor method (there can only be one) must be named
+    // `constructor`, and can take arguments like any other function.
+    constructor(name, animalMessage, canSwim) {
+        this.name = name;
+        this.#message = animalMessage;
+        // you can also define properties inside the constructor using `this`
+        this.canSwim = canSwim;
+    }
+}
+
+// you use the constructor to make a new object using the new keyword and the
+// class name
+let whale = new Animal("Whale", "Wooooo...", true);
+// these objects are just like any other objects, except they have the
+// properties we defined earlier
+whale.name === "Whale";
+whale.canSwim === true;
+// whale.#message === "Wooooo..."; <- not allowed, #message is private
+whale.sayMessage() === "Whale says: Wooooo...";
+
+// classes can extend other classes to represent a more specific type of
+// blueprint
+class Platypus extends Animal {
+    // subclasses can add their own methods and properties
+    layEggs() {
+        // TODO lay eggs
+    }
+
+    constructor() {
+        // you must call the parent class's constructor with the super keyword
+        super("Platypus", "Chatter", true);
+        // you can then access a parent class's properties and methods
+        this.laysEggs = true;
+        // not private ones though, those are still inaccessible
+        // this.#message = "Growl"; <- not allowed
+    }
+}
+
+let perry = new Platypus();
+// you can use parent class properties and methods on the subclass object
+perry.canSwim === true;
+perry.laysEggs === true;
+perry.sayMessage() === "Platypus says: Chatter"
+// and also use the new ones defined by the subclass
+perry.layEggs();
+```
+
+## Interacting with the DOM
+An important part of using JavaScript is changing and responding to changes in the DOM. The DOM, short for **D**ocument **O**bject **M**odel, is the browser's internal data structure for the contents of a page. Web browsers expose a number of JavaScript objects and functions to let you interact with this model. Understanding this section requires a basic understanding of HTML and CSS. Check out [our article on HTML](https://github.com/DevDogs-UGA/DevDogs-Academy/tree/feature/js-article/frontend/html) or [on CSS](https://github.com/DevDogs-UGA/DevDogs-Academy/tree/feature/js-article/frontend/css) if you need a quick intro (or refresher!).
```

Comment:
We'll need to come back and make sure that these links are correct once their respective articles get written. (cc #1, #2)

Reply:
Noted.